File: iPad/VerBank-CSTSv3-IOSClientApi/CSTS3Comm/comm/QuoteData.m
//
//  QuoteData.m
//  VerBank-CSTSv3-IOSClientApi
//
//  Created By Zhanglei on 15/6/30.
//  Copyright (c) 2015年 allone. All rights reserved.
//

#import "QuoteData.h"
static NSString * jsonId = @"QuoteData";

static NSString * instrument = @"1";
static NSString * bid = @"2";
static NSString * ask = @"3";
static NSString * quoteDay = @"4";
static NSString * quoteTime = @"5";
static NSString * openPrice = @"6";
static NSString * highPrice = @"7";
static NSString * lowPrice = @"8";
static NSString * yclosePrice = @"9";
static NSString * lastBid = @"10";
static NSString * tradeable = @"11";
static NSString * Close = @"12";

static NSString * bidBank = @"13";
static NSString * bidId = @"14";
static NSString * askBank = @"15";
static NSString * askId = @"16";

static NSString * lastAsk = @"17";
static NSString * bidAmount = @"18";
static NSString * askAmount = @"19";
@implementation QuoteData
- (id)init{
    self = [super init];
    if(self != nil)
    {
        [super setEntry:@"jsonId" entry:jsonId];
    }
    return self;
}
jsonImplementionString(Instrument,  instrument)
jsonImplementionDouble(Bid,         bid)
jsonImplementionDouble(Ask,         ask)
jsonImplementionString(QuoteDay,    quoteDay)
jsonImplementionLongLong(QuoteTime,     quoteTime)

jsonImplementionDouble(OpenPrice,   openPrice)
jsonImplementionDouble(HighPrice,   highPrice)
jsonImplementionDouble(LowPrice ,   lowPrice)
jsonImplementionDouble(YclosePrice ,yclosePrice)
jsonImplementionDouble(LastBid ,    lastBid)
jsonImplementionBoolean(Tradeable,  tradeable)
jsonImplementionBoolean(Close,      Close)

jsonImplementionString(BidBank,     bidBank)
jsonImplementionString(BidId ,      bidId)
jsonImplementionString(AskBank,     askBank)
jsonImplementionString(AskId,       askId)

jsonImplementionDouble(LastAsk ,    lastAsk)
jsonImplementionDouble(BidAmount,   bidAmount)
jsonImplementionDouble(AskAmount ,  askAmount)
@end

File: WeChatService/EX02-CSTSv3-ClientAPI/src/jedi/ex01/CSTS3/comm/struct/QuoteData.java
package jedi.ex01.CSTS3.comm.struct;
import allone.json.*;


public class QuoteData extends allone.json.AbstractJsonData {

	public static final String jsonId = "18";

	public static final String instrument = "1";
	public static final String bid = "2";
	public static final String ask = "3";
	public static final String quoteDay = "4";
	public static final String quoteTime = "5";
	public static final String openPrice = "6";
	public static final String highPrice = "7";
	public static final String lowPrice = "8";
	public static final String yclosePrice = "9";
	public static final String lastBid = "10";
	public static final String tradeable = "11";
	public static final String close = "12";
	public static final String tradeVolume = "13";
	public static final String lots = "14";

	public QuoteData(){
		super();
		setEntry("jsonId", jsonId);
	}

	public String getInstrument() {
		try {
			String data=getEntryString(QuoteData.instrument);
			return data;
		} catch (RuntimeException e) {
			return null;
		}
	}

	public void setInstrument(String instrument) {
		setEntry(QuoteData.instrument, instrument);
	}

	public double getBid() {
		try {
			double data=getEntryDouble(QuoteData.bid);
			return data;
		} catch (RuntimeException e) {
			return 0;
		}
	}

	public void setBid(double bid) {
		setEntry(QuoteData.bid, bid);
	}

	public double getAsk() {
		try {
			double data=getEntryDouble(QuoteData.ask);
			return data;
		} catch (RuntimeException e) {
			return 0;
		}
	}

	public void setAsk(double ask) {
		setEntry(QuoteData.ask, ask);
	}

	public String getQuoteDay() {
		try {
			String data=getEntryString(QuoteData.quoteDay);
			return data;
		} catch (RuntimeException e) {
			return null;
		}
	}

	public void setQuoteDay(String quoteDay) {
		setEntry(QuoteData.quoteDay, quoteDay);
	}

	public long getQuoteTime() {
		try {
			long data=getEntryLong(QuoteData.quoteTime);
			return data;
		} catch (RuntimeException e) {
			return 0;
		}
	}

	public void setQuoteTime(long quoteTime) {
		setEntry(QuoteData.quoteTime, quoteTime);
	}

	public double getOpenPrice() {
		try {
			double data=getEntryDouble(QuoteData.openPrice);
			return data;
		} catch (RuntimeException e) {
			return 0;
		}
	}

	public void setOpenPrice(double openPrice) {
		setEntry(QuoteData.openPrice, openPrice);
	}

	public double getHighPrice() {
		try {
			double data=getEntryDouble(QuoteData.highPrice);
			return data;
		} catch (RuntimeException e) {
			return 0;
		}
	}

	public void setHighPrice(double highPrice) {
		setEntry(QuoteData.highPrice, highPrice);
	}

	public double getLowPrice() {
		try {
			double data=getEntryDouble(QuoteData.lowPrice);
			return data;
		} catch (RuntimeException e) {
			return 0;
		}
	}

	public void setLowPrice(double lowPrice) {
		setEntry(QuoteData.lowPrice, lowPrice);
	}

	public double getYclosePrice() {
		try {
			double data=getEntryDouble(QuoteData.yclosePrice);
			return data;
		} catch (RuntimeException e) {
			return 0;
		}
	}

	public void setYclosePrice(double yclosePrice) {
		setEntry(QuoteData.yclosePrice, yclosePrice);
	}

	public double getLastBid() {
		try {
			double data=getEntryDouble(QuoteData.lastBid);
			return data;
		} catch (RuntimeException e) {
			return 0;
		}
	}

	public void setLastBid(double lastBid) {
		setEntry(QuoteData.lastBid, lastBid);
	}

	public boolean isTradeable() {
		try {
			boolean data=getEntryBoolean(QuoteData.tradeable);
			return data;
		} catch (RuntimeException e) {
			return false;
		}
	}

	public void setTradeable(boolean tradeable) {
		setEntry(QuoteData.tradeable, tradeable);
	}

	public boolean isClose() {
		try {
			boolean data=getEntryBoolean(QuoteData.close);
			return data;
		} catch (RuntimeException e) {
			return false;
		}
	}

	public void setClose(boolean close) {
		setEntry(QuoteData.close, close);
	}

	public long getTradeVolume() {
		try {
			long data=getEntryLong(QuoteData.tradeVolume);
			return data;
		} catch (RuntimeException e) {
			return 0;
		}
	}

	public void setTradeVolume(long tradeVolume) {
		setEntry(QuoteData.tradeVolume, tradeVolume);
	}

	public long getLots() {
		try {
			long data=getEntryLong(QuoteData.lots);
			return data;
		} catch (RuntimeException e) {
			return 0;
		}
	}

	public void setLots(long lots) {
		setEntry(QuoteData.lots, lots);
	}


}